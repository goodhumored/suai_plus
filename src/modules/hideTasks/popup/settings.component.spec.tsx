import { cleanup, fireEvent, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { IsHidingRepositoryMock } from "../../../__test__/__mocks__/isHidingRepository.mock";
import HideTasksSetting from "./setting.component";
jest.mock("../background/isHiding.repository", () => ({
  IsHidingRepository: IsHidingRepositoryMock
}));

describe("HideTasksSetting test suit", () => {
  afterEach(cleanup);
  describe("Rendering", () => {
    describe("should be unchecked", () => {
      it("renders well", () => {
        jest.spyOn(IsHidingRepositoryMock, "read").mockResolvedValue(false);
        const component = renderer.create(<HideTasksSetting />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
      it("componentDidMount should be called", () => {
        jest.spyOn(IsHidingRepositoryMock, "read").mockResolvedValue(false);
        const component = renderer.create(<HideTasksSetting />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
        expect(IsHidingRepositoryMock.read).toHaveBeenCalled();
      });
    });
    it("should be checked", () => {
      jest.spyOn(IsHidingRepositoryMock, "read").mockResolvedValue(true);
      const component = renderer.create(<HideTasksSetting />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
      expect(IsHidingRepositoryMock.read).toHaveBeenCalled();
    });
  });
  describe("click", () => {
    it("checkbox should work", () => {
      const { container } = render(<HideTasksSetting />);
      const checkbox = container.querySelector(
        "input#hideTasks"
      ) as HTMLInputElement;
      fireEvent.click(checkbox!);
      expect(checkbox?.checked).toBeTruthy();
    });
    it("should send new value to repository", () => {
      jest.spyOn(IsHidingRepositoryMock, "update");
      const { container } = render(<HideTasksSetting />);
      const checkbox = container.querySelector(
        "input#hideTasks"
      ) as HTMLInputElement;
      fireEvent.click(checkbox!);
      expect(checkbox?.checked).toBeTruthy();
      expect(IsHidingRepositoryMock.update).toHaveBeenLastCalledWith(true);
      fireEvent.click(checkbox!);
      expect(checkbox?.checked).toBeFalsy();
      expect(IsHidingRepositoryMock.update).toHaveBeenLastCalledWith(false);
    });
  });
});
