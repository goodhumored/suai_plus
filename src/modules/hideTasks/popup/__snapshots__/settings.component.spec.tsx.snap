// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HideTasksSetting test suit Rendering should be checked 1`] = `
<div>
  <input
    checked={false}
    id="hideTasks"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    htmlFor="hideTasks"
  >
    Скрывать отмеченные задачи
  </label>
</div>
`;

exports[`HideTasksSetting test suit Rendering should be unchecked componentDidMount should be called 1`] = `
<div>
  <input
    checked={false}
    id="hideTasks"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    htmlFor="hideTasks"
  >
    Скрывать отмеченные задачи
  </label>
</div>
`;

exports[`HideTasksSetting test suit Rendering should be unchecked renders well 1`] = `
<div>
  <input
    checked={false}
    id="hideTasks"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    htmlFor="hideTasks"
  >
    Скрывать отмеченные задачи
  </label>
</div>
`;
